mutation-testing/pr-differences/action.yml

```diff
@@ -0,0 +1,173 @@
+          else
+            echo "$regex_pattern" > mutants_by_packages/regex_small.txt
+          fi
+        done
+
+        exit 0
+      working-directory: actions-repo/mutation-testing/shell-scripts
+
+    - name: Run mutants
+      shell: bash
+      run: |
+        # Disable immediate exit on error
+        set +e
+
+        # Check which type of mutants to run: big, small, with or without shards, then capture the exit code
+        if [[ ${{ inputs.shard }} == -1 ]]; then
+          if [[ ${{ inputs.package-dimension }} == big ]]; then
+            if [[ -f mutants_by_packages/regex_big.txt ]]; then
+              echo "Running mutants without shards on big packages"
+              cargo mutants --no-shuffle -vV -F "$(<mutants_by_packages/regex_big.txt)" --output ./ -- --all-targets -- --test-threads 1
+              exit_code=$?
+            else
+              echo "File containing mutants doesn't exist!"
+            fi
+          else
+            if [[ -f mutants_by_packages/regex_small.txt ]]; then
+              echo "Running mutants without shards on small packages"
+              cargo mutants --no-shuffle -vV -F "$(<mutants_by_packages/regex_small.txt)" --output ./ -- --all-targets
+              exit_code=$?
+            else
+              echo "File containing mutants doesn't exist!"
+            fi
+          fi
+        else
+          if [[ ${{ inputs.package-dimension }} == big ]]; then
+            if [[ -f mutants_by_packages/regex_big.txt ]]; then
+              echo "Running mutants with shard ${{ inputs.shard }} on big packages"
+              cargo mutants --no-shuffle -vV -F "$(<mutants_by_packages/regex_big.txt)" --shard ${{ inputs.shard }}/8 --output ./ -- --all-targets -- --test-threads 1
+              exit_code=$?
+            else
+              echo "File containing mutants doesn't exist!"
+            fi
+          else
+            if [[ -f mutants_by_packages/regex_small.txt ]]; then
+              echo "Running mutants with shard ${{ inputs.shard }} on small packages"
+              cargo mutants --no-shuffle -vV -F "$(<mutants_by_packages/regex_small.txt)" --shard ${{ inputs.shard }}/4 --output ./ -- --all-targets
+              exit_code=$?
+            else
+              echo "File containing mutants doesn't exist!"
+            fi
+          fi
+        fi
+
+        # Create the folder only containing the outcomes (.txt files) and make a file containing the exit code of the command
+        mkdir mutants-shard-${{ inputs.package-dimension }}-${{ inputs.shard }}
+        echo "$exit_code" > ./mutants-shard-${{ inputs.package-dimension }}-${{ inputs.shard }}/exit_code.txt
+        mv ./mutants.out/*.txt mutants-shard-${{ inputs.package-dimension }}-${{ inputs.shard }}/
+
+        # Enable immediate exit on error again
+        set -e
+      working-directory: actions-repo/mutation-testing/shell-scripts
+
+    - name: Upload artifact
+      uses: actions/upload-artifact@v3
+      if: always()
```

Comment:
what would happen here if the path doesn't exist?
i'm not  a fan of using `always()`, have you tried alternatives like a conditional of previous steps?

Reply:
Removed the `always()` - we don't need to upload an artifact if the previous steps failed, and we are also exiting early if something happens anyway.

Reply:
Not seeing the change yet, but will resolve once it's commited

Reply:
I’ve done the changes on a different branch and tested them there https://github.com/ASuciuX/stacks-core/pull/47. Did this in order to keep this cleaner and the commits on this as well. Please let me know if that’s alright.